kotlinx-coroutines-core/common/src/channels/Channel.kt

```diff
@@ -673,41 +1163,46 @@ public interface ChannelIterator<out E> {
  * With a single-threaded [dispatcher][CoroutineDispatcher] like [Dispatchers.Main], this gives a
  * guarantee that the coroutine promptly reacts to the cancellation of its [Job] and does not resume its execution.
  *
- * > **Prompt cancellation guarantee** for channel operations was added since `kotlinx.coroutines` version `1.4.0`
- * > and had replaced a channel-specific atomic-cancellation that was not consistent with other suspending functions.
- * > The low-level mechanics of prompt cancellation are explained in [suspendCancellableCoroutine] function.
+ * > **Prompt cancellation guarantee** for channel operations was added in `kotlinx.coroutines` version `1.4.0`
+ * > and has replaced the channel-specific atomic cancellation that was not consistent with other suspending functions.
+ * > The low-level mechanics of prompt cancellation are explained in the [suspendCancellableCoroutine] documentation.
  *
  * ### Undelivered elements
  *
  * As a result of the prompt cancellation guarantee, when a closeable resource
- * (like open file or a handle to another native resource) is transferred via a channel from one coroutine to another,
- * it can fail to be delivered and will be lost if the receiving operation is cancelled in transit.
+ * (like an open file or a handle to another native resource) is transferred via a channel,
+ * it can be successfully extracted from the channel,
+ * but still be lost if the receiving operation is cancelled in parallel.
  *
- * A `Channel()` constructor function has an `onUndeliveredElement` optional parameter.
- * When `onUndeliveredElement` parameter is set, the corresponding function is called once for each element
+ * The `Channel()` constructor function has the optional parameter `onUndeliveredElement`.
```

Comment:
```suggestion
 * The `Channel()` factory function has the optional parameter `onUndeliveredElement`.
```
It's called this way everywhere else.